File: src/de/saschahlusiak/freebloks/network/NET_GRANT_PLAYER.java
package de.saschahlusiak.freebloks.network;

import java.io.ByteArrayOutputStream;

public class NET_GRANT_PLAYER extends NET_HEADER {
	public int player; /* int 8 */
		
	public NET_GRANT_PLAYER() {
		super(Network.MSG_GRANT_PLAYER, 1);
	}
	
	public NET_GRANT_PLAYER(NET_HEADER from) {
		super(from);
		player = buffer[0];
	}
	
	@Override
	void prepare(ByteArrayOutputStream bos) {
		super.prepare(bos);
		bos.write(player);
	}
}

File: src/main/java/de/saschahlusiak/freebloks/network/message/NetGrantPlayer.kt
package de.saschahlusiak.freebloks.network.message

import de.saschahlusiak.freebloks.network.*
import java.nio.ByteBuffer

data class NetGrantPlayer(val player: Int): Message(Network.MSG_GRANT_PLAYER, 1) {
    init {
        assert(player in 0..3) { "Invalid player $player" }
    }

    override fun write(buffer: ByteBuffer) {
        buffer.put(header)
        buffer.put(player.toByte())
    }

    companion object {
        fun from(data: ByteBuffer): NetGrantPlayer {
            val player = data.get().toInt()
            return NetGrantPlayer(player)
        }
    }
}